superset/db_engine_specs/gsheets.py

```diff
@@ -57,7 +71,7 @@ class GSheetsParametersSchema(Schema):
 
 class GSheetsParametersType(TypedDict):
     service_account_info: str
-    catalog: Optional[dict[str, str]]
+    catalog: dict[str, str] | None
```

Comment:
Just curious why moving from `Optional` ? Because I see `Optional` being used in several other `TypedDict` classes. Want to confirm if this is a pattern we aim to follow moving forward, thanks.

Reply:
We bumped the minimum Python version we support in Superset (I think to 3.9?), and in it we can use the syntax `foo | None` instead of `Optional[foo]`. This way we don't need to import `Optional`, and the type declaration looks more natural (for example, it's similar to Typescript's `foo | null` for an optional `foo`).

Reply:
Oh cool! Good to know moving forward then. Thanks @betodealmeida